src/metax_api/services/catalog_record_service.py

```diff
@@ -52,61 +52,90 @@ def get_queryset_search_params(cls, request):
 
-        if request.query_params.get('metadata_owner_org', False):
-            queryset_search_params['metadata_owner_org__in'] = request.query_params['metadata_owner_org'].split(',')
+            if request.query_params.get('contract_org_identifier', False):
+                if request.user.username not in ('metax', 'tpas'):
+                    raise Http403({ 'detail': ['query parameter pas_filter is restricted']})
+                queryset_search_params['contract__contract_json__organization__organization_identifier__iregex'] = \
+                    request.query_params['contract_org_identifier']
 
-        if request.query_params.get('contract_org_identifier', False):
-            if request.user.username not in ('metax', 'tpas'):
-                raise Http403({ 'detail': ['query parameter pas_filter is restricted']})
-            queryset_search_params['contract__contract_json__organization__organization_identifier__iregex'] = \
-                request.query_params['contract_org_identifier']
+            if request.query_params.get('pas_filter', False):
+                cls.set_pas_filter(queryset_search_params, request)
 
-        if request.query_params.get('pas_filter', False):
-            cls.set_pas_filter(queryset_search_params, request)
+            if CommonService.has_research_agent_query_params(request):
+                cls.set_actor_filters(queryset_search_params, request)
 
-        if CommonService.has_research_agent_query_params(request):
-            cls.set_actor_filters(queryset_search_params, request)
+            if request.query_params.get('data_catalog', False):
+                queryset_search_params['data_catalog__catalog_json__identifier__iregex'] = \
+                    request.query_params['data_catalog']
 
-        if request.query_params.get('data_catalog', False):
-            queryset_search_params['data_catalog__catalog_json__identifier__iregex'] = \
-                request.query_params['data_catalog']
+            return queryset_search_params
+
+    @staticmethod
+    def filter_by_state(request, queryset_search_params):
+        '''
+        Helper method to filter returning data by state: unauthenticated
+        users get only published data, and end users get only published &
+        their own drafts
+        '''
+        state_filter = None
+
+        if request.user.username is None: # unauthenticated user
+            state_filter = Q(state='published')
+        elif request.user.is_service:  # service account
+            pass
+        else: # enduser api
+            state_filter = Q(state='published') | Q(state='draft', metadata_provider_user=request.user.username)
+
+        #if state_filter:
+        if 'q_filters' in queryset_search_params:
+            queryset_search_params['q_filters'].append(state_filter)
+        elif state_filter is None:
+            queryset_search_params = {}
+        else:
+            queryset_search_params['q_filters'] = [state_filter]
```

Comment:
Once the code that calls this method is changed to pass the variable ``queryset_search_params`` as an argument to this method, then this block should use the commented line ``if state_filter:``, and the middle middle ``elif...`` block should be removed (this method should not reset the ``queryset_search_params`` back to empty dict)